diff --git a/Dockerfile b/Dockerfile
index 45e4bda5..070793f3 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -194,6 +194,7 @@ FROM --platform=${BUILDARCH} node:lts AS frontend-build
 ARG BUILDARCH
 ARG BUILD_FRONTEND=0
 ARG RUNTIME_UID
+ARG RUNTIME_GID
 
 ARG NODE_ENV=production
 ENV NODE_ENV=$NODE_ENV
@@ -261,9 +262,8 @@ USER ${RUNTIME_UID}:${RUNTIME_GID}
 # NOTE: we skip the autoloader generation here since we don't have all files avaliable (yet)
 RUN --mount=type=cache,id=pixelfed-composer-${PHP_VERSION},sharing=locked,uid=${RUNTIME_UID},gid=${RUNTIME_GID},target=/cache/composer \
     --mount=type=bind,source=composer.json,target=/var/www/composer.json \
-    --mount=type=bind,source=composer.lock,target=/var/www/composer.lock \
     set -ex \
-    && composer install --prefer-dist --no-autoloader --ignore-platform-reqs --no-scripts
+    && composer update --prefer-dist --no-autoloader --ignore-platform-reqs --no-scripts
 
 # Copy all other files over
 COPY --chown=${RUNTIME_UID}:${RUNTIME_GID} . /var/www/
