From 1309db17722836229b68c370613742a5f39664ff Mon Sep 17 00:00:00 2001
From: Anil Kulkarni <anil@terminal.space>
Date: Sun, 12 Jan 2025 13:29:13 -0800
Subject: [PATCH] Add a push notification when a user is mentioned in a comment

---
 app/Jobs/MentionPipeline/MentionPipeline.php | 27 +++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/app/Jobs/MentionPipeline/MentionPipeline.php b/app/Jobs/MentionPipeline/MentionPipeline.php
index cec42f09..14c713a7 100644
--- a/app/Jobs/MentionPipeline/MentionPipeline.php
+++ b/app/Jobs/MentionPipeline/MentionPipeline.php
@@ -5,12 +5,18 @@ namespace App\Jobs\MentionPipeline;
 use App\Mention;
 use App\Notification;
 use App\Status;
+use App\User;
 use Illuminate\Bus\Queueable;
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Foundation\Bus\Dispatchable;
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Queue\SerializesModels;
+use Log;
+use App\Jobs\PushNotificationPipeline\MentionPushNotifyPipeline;
+use App\Services\NotificationAppGatewayService;
+use App\Services\PushNotificationService;
 use App\Services\StatusService;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 
 class MentionPipeline implements ShouldQueue
 {
@@ -49,6 +55,8 @@ class MentionPipeline implements ShouldQueue
         $actor = $this->status->profile;
         $target = $this->mention->profile_id;
 
+        Log::info('MentionPipeline: 1');
+
         $exists = Notification::whereProfileId($target)
                   ->whereActorId($actor->id)
                   ->whereIn('action', ['mention', 'comment'])
@@ -57,9 +65,11 @@ class MentionPipeline implements ShouldQueue
                   ->count();
 
         if ($actor->id === $target || $exists !== 0) {
-            return true;
+            Log::info('MentionPipeline: 2' . $actor->id . ' ' . $target . ' ' . $exists);
+            return;
         }
 
+        Log::info('MentionPipeline: 3');
         Notification::firstOrCreate(
             [
                 'profile_id' => $target,
@@ -71,5 +81,20 @@ class MentionPipeline implements ShouldQueue
         );
 
         StatusService::del($status->id);
+
+        if (NotificationAppGatewayService::enabled()) {
+            Log::info('MentionPipeline: 4');
+            if (PushNotificationService::check('mention', $target)) {
+                Log::info('MentionPipeline: 5');
+                $user = User::whereProfileId($target)->first();
+                if ($user && $user->expo_token && $user->notify_enabled) {
+                    Log::info('MentionPipeline: 6');
+                    MentionPushNotifyPipeline::dispatch($user->expo_token, $actor->username)->onQueue('pushnotify');
+                    Log::info('MentionPipeline: 7');
+                } else {
+                    Log::info('MentionPipeline: 8');
+                }
+            }
+        }
     }
 }
-- 
2.48.0

